### DISCUSSION TITLE
Extension of Password Token Expiry Time - UAA Version 4.7.0

### DISCUSSION THREAD

**Discussion by kguptaumesh** [cloudfoundry/uaa#2398] [Q&A]
Hi All,

Currently, we are working to extend the password token expiry time (in our custom code) from 30 minutes to 1 hour.
However, however when token is used after 30 minutes to set the password, it fails in UAA.

endpoint used - POST http://localhost:8080/uaaz/password_change HTTP/1.1

Version used - https://docs.cloudfoundry.org/api/uaa/version/4.7.0/

When I was exploring the document, I have located a hardcoded proeprty in UaaResetPasswordService class.

 public static final int PASSWORD_RESET_LIFETIME = 30 * 60 * 1000;

Hence, I am curious to know if this is the value which is stopping me to extend the expiry time.  I am hopeful to have a good discussion around this and extract an answer of my query.

Screen shot of the UAA hard coded value. 
![image](https://github.com/cloudfoundry/uaa/assets/63576609/901ea37e-519e-4b9d-8e8c-37c100dca764)

Regards,
Umesh

**Comment by strehle** [comment]
Hi @kguptaumesh,  

you use a custom UAA or a fork? 

If you ask to extend the time with a configuration option, then yes this is the calculation for expiry of the password reset.

If you have a proposal for a configuration option, then please open a PR to provide you changes 

regards

**Reply by kguptaumesh** [reply]
Thank you @strehle for your response.  
we use custom UAA but small customization. 

Yeah if it is configurable , this should serve the purpose.  
If I attempt to change 4.7.0 version code to raise PR, will it help? or there is any process defined around this?

Regards,
Umesh
